### PR TITLE
Masasa_create_badge_permission

### PR DESCRIPTION
**Pull request by ThapeloMasasa**
# Description
This PR enables the user to have access to badges and have access to creating padges in the backend,  to enable the user to have the ability to create badges.
Screenshot 2023-09-09 at 9 25 22 PM
Screenshot 2023-09-09 at 9 29 45 PM

## Related PRS (if any):
To test this backend PR you need to checkout the [#1287](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1287) frontend PR.
…

## Main changes explained:
-added  individualpermission check to postbadges and view badges in badgesController
…

## How to test:
1. Follow the test instructions mentioned on the frontend

### CODE REVIEW

**Review by navneeeth — APPROVED** [review]
Hello, @ThapeloMasasa! I have reviewed your changes and they work as intended. Good implementation! I have commented a full review on the [frontend PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1287#pullrequestreview-1623199960).

**Review by DavidC0126 — COMMENTED** [review]
Tested the PR with FE PR#1287, left the comment on FE PR.

**Comment by Shiwani99** [comment]
Hi,
I tested the PR and I have left my comments at PR#1287

**Review by vikrambadhan — APPROVED** [review]
Hi, I tested your PR with FR #1287 and works as intended!

**Comment by wantingxu7** [comment]
Tested with FE #1287 and comment there

**Comment by TuanDinh140194** [comment]
Hi! I have tested your PR#1287 function and review there

**Review by robertoooc — APPROVED** [review]
Tested along with FE 1287, left comments on other PR.

**Comment by Chris33871** [comment]
Hi @ThapeloMasasa , 

I left a comment on the Front End post

**Review by StrawberryCalpico — APPROVED** [review]
I reviewed your PR, the details of test: [FE #1287](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1287#pullrequestreview-1677724666)

**Review by malikjahanzaib — APPROVED** [review]
Hi @ThapeloMasasa, I have reviewed this PR and everything seems to work as described. Detailed comments on FE #1287.

**Comment by RheaWu1212** [comment]
Hi! I have tested your PR#1287 function and review there

File: src/controllers/badgeController.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 const mongoose = require('mongoose');
 const UserProfile = require('../models/userProfile');
-const { hasPermission } = require('../utilities/permissions');
+const { hasPermission, hasIndividualPermission } = require('../utilities/permissions');
 const escapeRegex = require('../utilities/escapeRegex');
 
 const badgeController = function (Badge) {
   const getAllBadges = async function (req, res) {
-    if (!await hasPermission(req.body.requestor.role, 'seeBadges')) {
+    if (!await hasPermission(req.body.requestor.role, 'seeBadges') && !await hasIndividualPermission(req.body.requestor.requestorId, 'seeBadgeManagementTab')) {
```
Comment (nathanah) [inline src/controllers/badgeController.js:8]:
```suggestion
    if (!await hasPermission(req.body.requestor, 'seeBadges') && !await hasPermission(req.body.requestor, 'seeBadgeManagementTab')) {
```
Probably need to pull the latest changes. The arguments have changed.

File: src/controllers/badgeController.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ const badgeController = function (Badge) {
   };
 
   const postBadge = async function (req, res) {
-    if (!await hasPermission(req.body.requestor.role, 'createBadges')) {
+    if (!await hasPermission(req.body.requestor.role, 'createBadges') && !await hasIndividualPermission(req.body.requestor.requestorId, 'seeBadgeManagementTab')) {
```
Comment (nathanah) [inline src/controllers/badgeController.js:60]:
```suggestion
    if (!await hasPermission(req.body.requestor, 'createBadges') && !await hasPermission(req.body.requestor, 'seeBadgeManagementTab')) {
```
Probably need to pull the latest changes. The arguments have changed.

**Review by nathanah — CHANGES_REQUESTED** [review]
I don't think `seeBadgeManagementTab` should exist as a permission as it's just a combination of `seeBadges` and `createBadges`. However, if you are to make it, then you will also need to add it to `createInitialPermissions.js` under every role that should have it.

**Review by shubhankarval — APPROVED** [review]
Hey @ThapeloMasasa, left a comment on FE PR [#1287](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1287).

**Review by Pandani07 — APPROVED** [review]
This PR works as intended along with the frontend PR, left a review there.

**Review by skaush21 — COMMENTED** [review]
comment on frontend [PR 1287](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1287)

**Pull request closed without merging**